plugin/pom.xml

```diff
@@ -134,6 +134,14 @@
           <failOnError>false</failOnError>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.jenkins-ci.tools</groupId>
+        <artifactId>maven-hpi-plugin</artifactId>
+        <configuration>
+          <!-- avoid snakeyaml classpath conflicts -->
+          <pluginFirstClassLoader>true</pluginFirstClassLoader>
```

Comment:
instead of this. I suggest you create an hpi for snakeyaml just like [apache httpcomponents](https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin) or [handy uri templates](https://github.com/jenkinsci/handy-uri-templates-2-api-plugin)

Reply:
It will not help to prevent compatibility issues on updates, unfortunately

Reply:
Nothing prevents compatibility when devs don't follow some convention. However it will help with the plugin first class loader 👍

Reply:
Oh it's that bad 😭